src/stk/io/nifti.cpp

```diff
@@ -295,39 +287,51 @@ namespace nifti {
         // Only supported modes for now
         nhdr.qform_code = NIFTI_XFORM_SCANNER_ANAT;
         nhdr.sform_code = NIFTI_XFORM_UNKNOWN;
-        
+
         // Set according to method 2 (nifti1.h)
-         
-        float3 origin = vol.origin();
 
-        mat44 mat = nifti_make_orthog_mat44(-1, 0, 0,
-                                            0, 1, 0,
-                                            0, 0, 1);
+        const Matrix3x3f& d = vol.direction();
+        mat44 mat = nifti_make_orthog_mat44(
```

Comment:
If the direction matrix is not orthogonal, we lose information here. It is a limitation of NIFTI compared to other formats. Even though it is probably not very sane to use a non-orthogonal direction matrix...